### ISSUE TITLE
Switch Pro Controller not registering on 2.1.4

### ISSUE THREAD

**Issue by SenerioIX** [visualboyadvance-m/visualboyadvance-m#629]
### Expected behavior
Switch Pro Controller input mapping to work

### Actual behavior
Switch Pro Controller input mapping doesn't work

### Extra Info
I launch 2.0.0, go into inputs and can normally bind the controller buttons.
I launch 2.1.4 and do the same thing but it doesn't seem to register anything.

### Steps to reproduce the behavior
Plug in Switch Pro Controller, start VBA, try setting the inputs, nothing happens.

### Build details

OS:
Windows 10

Emulator version : 
Latest (v2.1.4)

### Hardware details:

Wired Switch Pro Controllers, Windows in-built drivers.

**Comment by ZachBacon** [comment]
This is most likely due to the fact it's not using xinput. I do have a pro controller and a controller that has a switch input emulation, so I'll give that a local test and see what's going on, but my guess is that it isn't using xinput. I'll report what I find.

**Comment by rkitover** [comment]
Please try this build which has the new joystick code:

https://github.com/visualboyadvance-m/visualboyadvance-m/issues/354#issuecomment-582162781

**Comment by rkitover** [comment]
Also when you try that build, please also check that the default mappings are correct. That was one of the goals of my joystick refactor.

**Comment by ZachBacon** [comment]
@rkitover SDL 2.0.12 recently added some new API something like SDL_GameControllerTypeForIndex(), SDL_GameControllerGetType() to return the type of a game controller (Xbox 360, Xbox One, PS3, PS4, or Nintendo Switch Pro) so worse comes to worst, could add conditionals for switch controllers

**Comment by SenerioIX** [comment]
Nope. Still not registering anything and I'm always counter validating with 2.0.0 so you can be sure I didn't mess up the test.

**Comment by rkitover** [comment]
Related: #631 

So looks like switch controllers don't work as a GameController in SDL at all.

I will add regular joystick fallback code.

**Comment by Squall-Leonhart** [comment]
working via sdl3
